package.json

```diff
@@ -2,25 +2,89 @@
     "dist"
   ],
+  "engines": {
+    "node": ">=14"
+  },
   "scripts": {
+    "add-contributors": "all-contributors add",
+    "build": "rollup -c",
+    "clean": "rimraf dist demo/dist",
+    "deploy": "gh-pages -d demo/",
+    "format": "prettier --write \"**/*.+(js|json|yml|yaml|css|md|ts)\"",
+    "lint": "eslint \"./**/*.{js,ts}\"",
+    "lint:fix": "eslint \"./**/*.{js,ts}\" --fix",
+    "postbuild": "copyfiles dist/** demo",
+    "prebuild": "yarn clean",
+    "prepublishOnly": "yarn build",
+    "serve": "serve demo",
+    "start": "concurrently \"yarn watch\" \"yarn serve\"",
     "test": "jest",
     "test:coverage": "yarn test --coverage --collectCoverageFrom=src/**/*js --collectCoverageFrom=!src/d3.js",
-    "test:staged": "yarn test --findRelatedTests",
     "test:coveralls": "yarn test --coverage && coveralls < coverage/lcov.info",
-    "clean": "rimraf dist",
-    "prebuild": "yarn clean",
-    "build": "rollup -c",
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix",
-    "format": "prettier --write \"**/*.+(js|json|yml|yaml|css|md)\"",
-    "deploy": "gh-pages -d demo/",
-    "prepublishOnly": "yarn build"
+    "test:staged": "yarn test --findRelatedTests",
+    "watch": "rollup -c -w --watch.onEnd=\"yarn postbuild\""
+  },
+  "devDependencies": {
+    "@babel/core": "^7.9.6",
+    "@babel/plugin-transform-modules-commonjs": "^7.9.6",
+    "@babel/preset-env": "^7.9.6",
+    "@rollup/plugin-babel": "^5.3.1",
+    "@rollup/plugin-commonjs": "^22.0.2",
+    "@rollup/plugin-node-resolve": "^14.1.0",
+    "@rollup/plugin-typescript": "^8.5.0",
+    "@types/d3-array": "^2.12.3",
+    "@types/d3-axis": "^1.0.16",
+    "@types/d3-drag": "^1.2.5",
+    "@types/d3-scale": "^3.3.2",
+    "@types/d3-selection": "^1.4.3",
+    "@types/d3-shape": "^1.3.8",
+    "@types/event-emitter-es6": "^1.1.1",
+    "@types/jest": "^26.0.24",
+    "all-contributors-cli": "^6.21.0",
+    "autoprefixer": "^9.8.0",
+    "babel-jest": "^26.6.3",
+    "concurrently": "^7.4.0",
+    "copyfiles": "^2.4.1",
+    "coveralls": "^3.1.0",
+    "cssnano": "^4.1.10",
+    "eslint": "8.24.0",
+    "eslint-config-ns-ts-base": "^3.3.0",
+    "gh-pages": "^2.2.0",
+    "husky": "^4.2.5",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^26.6.3",
+    "lint-staged": "^10.2.2",
+    "postcss": "^8.4.16",
+    "prettier": "^2.7.1",
+    "react-test-renderer": "^16.13.1",
+    "rimraf": "^3.0.2",
+    "rollup": "^2.79.1",
+    "rollup-plugin-bundle-size": "^1.0.3",
+    "rollup-plugin-multi-input": "^1.3.1",
+    "rollup-plugin-postcss": "^4.0.2",
+    "rollup-plugin-terser": "^7.0.2",
+    "rollup-plugin-visualizer": "^5.8.2",
+    "serve": "^14.0.1",
+    "ts-essentials": "^9.3.0",
+    "ts-jest": "26.5.6",
+    "typescript": "^4.8.3"
+  },
+  "dependencies": {
+    "d3-array": "2.4.0",
+    "d3-axis": "1.0.12",
+    "d3-drag": "1.2.5",
+    "d3-scale": "3.2.1",
+    "d3-selection": "1.4.1",
+    "d3-shape": "1.3.7",
+    "event-emitter-es6": "1.1.5"
   },
```

Comment:
I suggest using a fixed version here to make results reproducible when running the app in production. Newer versions of the `d3-*` packages are not yet compatible. This happens when I update all to the latest minor version and drag an item.

<img width="1832" alt="image" src="https://user-images.githubusercontent.com/1043668/192640193-41f96219-e91d-43e6-a18d-4b49b18da742.png">

Reply:
Btw, I've just tried updating all `d3` dependencies to the latest version:

```
"d3-array": "^3.2.0",
"d3-axis": "^3.0.0",
"d3-drag": "^3.0.0",
"d3-scale": "^4.0.2",
"d3-selection": "3.0.0",
"d3-shape": "^3.1.0"
```

and then this happens:

![Screen Capture on 2022-09-28 at 10-02-15](https://user-images.githubusercontent.com/1043668/192723400-f5162273-a749-4dd7-bef1-e2e93eac9cea.gif)

For whatever reason, the position is reset after each drag. This should be part of another PR. It's only a preview of what's probably next on the Todo list.

Reply:
Yup that `drag` function is nasty I remember that I spent countless hours trying to make it work, as long as it works using older version of d3 so be it :)